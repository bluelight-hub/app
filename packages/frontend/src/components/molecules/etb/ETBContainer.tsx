import { useEinsatztagebuch } from '@/hooks/etb/useEinsatztagebuch';
import { logger } from '@/utils/logger';
import { Form, Input, Modal } from 'antd';
import { useState } from 'react';
import { ETBTable } from '../../organisms/etb/ETBTable';

/**
 * Props für die ETBContainer-Komponente
 */
interface ETBContainerProps {
    /**
     * Gibt an, ob automatische Aktualisierungen aktiviert werden sollen
     */
    enableAutoRefresh?: boolean;
    /**
     * Intervall für automatische Aktualisierungen in Millisekunden
     */
    refreshInterval?: number;
}

/**
 * Container-Komponente für das Einsatztagebuch
 * Verwaltet die Datenabfrage und Paginierung
 */
export const ETBContainer: React.FC<ETBContainerProps> = ({
    enableAutoRefresh = true,
    refreshInterval = 15000, // 15 Sekunden
}) => {
    // Paginierungs-Status
    const [page, setPage] = useState(1);
    const [pageSize, setPageSize] = useState(10);

    // Überschreiben-Modal Status
    const [isUeberschreibenModalVisible, setIsUeberschreibenModalVisible] = useState(false);
    const [currentEntry, setCurrentEntry] = useState<{ id: string; beschreibung: string } | null>(null);
    const [form] = Form.useForm();

    // Hook für die Einsatztagebuch-Daten
    const {
        einsatztagebuch,
        archiveEinsatztagebuchEintrag,
        ueberschreibeEinsatztagebuchEintrag
    } = useEinsatztagebuch({
        filterParams: { page, limit: pageSize },
        refetchInterval: enableAutoRefresh ? refreshInterval : undefined
    });

    /**
     * Handler für Seitenwechsel
     */
    const handlePageChange = async (newPage: number, newPageSize: number) => {
        logger.info('Seitenwechsel angefordert:', { newPage, newPageSize, currentPage: page, currentPageSize: pageSize });

        try {
            // Nur API-Aufruf wenn sich die Werte geändert haben
            if (newPage !== page || newPageSize !== pageSize) {
                await einsatztagebuch.changePage(newPage, newPageSize);
                setPage(newPage);
                setPageSize(newPageSize);
                logger.info('Seitenwechsel abgeschlossen:', { page: newPage, pageSize: newPageSize });
            }
        } catch (error) {
            logger.error('Fehler beim Seitenwechsel:', error);
        }
    };

    /**
     * Handler für das Archivieren eines Eintrags
     */
    const handleArchiveEntry = (nummer: number) => {
        archiveEinsatztagebuchEintrag.mutate({ nummer }, {
            onSuccess: () => {
                logger.info(`Eintrag ${nummer} erfolgreich archiviert`);
                einsatztagebuch.refetch();
            },
            onError: (error) => {
                logger.error(`Fehler beim Archivieren des Eintrags ${nummer}:`, error);
            }
        });
    };

    /**
     * Öffnet das Modal zum Überschreiben eines Eintrags
     */
    const handleUeberschreibenOpen = (entry: { id: string; beschreibung: string }) => {
        setCurrentEntry(entry);
        form.setFieldsValue({ beschreibung: entry.beschreibung });
        setIsUeberschreibenModalVisible(true);
    };

    /**
     * Speichert die Überschreibung eines Eintrags
     */
    const handleUeberschreibenSubmit = () => {
        if (!currentEntry) return;

        form.validateFields().then(values => {
            ueberschreibeEinsatztagebuchEintrag.mutate({
                id: currentEntry.id,
                beschreibung: values.beschreibung
            }, {
                onSuccess: () => {
                    logger.info(`Eintrag ${currentEntry.id} erfolgreich überschrieben`);
                    setIsUeberschreibenModalVisible(false);
                    setCurrentEntry(null);
                    form.resetFields();
                    einsatztagebuch.refetch();
                },
                onError: (error) => {
                    logger.error(`Fehler beim Überschreiben des Eintrags:`, error);
                }
            });
        });
    };

    /**
     * Schließt das Modal zum Überschreiben eines Eintrags
     */
    const handleUeberschreibenCancel = () => {
        setIsUeberschreibenModalVisible(false);
        setCurrentEntry(null);
        form.resetFields();
    };

    // Debugging-Informationen
    logger.debug('ETBContainer Render:', {
        dataState: {
            items: einsatztagebuch.data.items.length,
            pagination: einsatztagebuch.data.pagination,
            loading: einsatztagebuch.query.isLoading,
            fetching: einsatztagebuch.query.isFetching,
        },
        componentState: {
            page,
            pageSize,
            isModalVisible: isUeberschreibenModalVisible,
        }
    });

    return (
        <div className="etb-container">
            <ETBTable
                entries={einsatztagebuch.data.items}
                isLoading={einsatztagebuch.query.isLoading || einsatztagebuch.query.isFetching}
                isEditing={ueberschreibeEinsatztagebuchEintrag.isPending}
                onArchiveEntry={handleArchiveEntry}
                onUeberschreibeEntry={(entry) => handleUeberschreibenOpen({ id: entry.id, beschreibung: entry.beschreibung })}
                pagination={einsatztagebuch.data.pagination ? {
                    ...einsatztagebuch.data.pagination,
                    hasNextPage: einsatztagebuch.data.pagination.currentPage < einsatztagebuch.data.pagination.totalPages,
                    hasPreviousPage: einsatztagebuch.data.pagination.currentPage > 1
                } : undefined}
                onPageChange={handlePageChange}
            />

            <Modal
                title="Eintrag überschreiben"
                open={isUeberschreibenModalVisible}
                onOk={handleUeberschreibenSubmit}
                onCancel={handleUeberschreibenCancel}
                confirmLoading={ueberschreibeEinsatztagebuchEintrag.isPending}
            >
                <Form form={form} layout="vertical">
                    <Form.Item
                        name="beschreibung"
                        label="Beschreibung"
                        rules={[{ required: true, message: 'Bitte geben Sie eine Beschreibung ein' }]}
                    >
                        <Input.TextArea rows={4} />
                    </Form.Item>
                </Form>
            </Modal>
        </div>
    );
}; 