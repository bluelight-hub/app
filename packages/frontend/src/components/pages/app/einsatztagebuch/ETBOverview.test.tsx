import { EtbEntryDto, EtbEntryDtoStatusEnum } from '@bluelight-hub/shared/client/models/EtbEntryDto';
import { QueryClient } from '@tanstack/react-query';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import { useEinsatztagebuch } from '../../../../hooks/etb/useEinsatztagebuch';
import { useFahrzeuge } from '../../../../hooks/etb/useFahrzeuge';
import { ETBOverview } from './ETBOverview';

// Mock-Daten für einen Eintrag
const mockEntryData = {
    id: '1',
    laufendeNummer: 1,
    timestampErstellung: new Date('2023-01-01T10:00:00'),
    timestampEreignis: new Date('2023-01-01T10:00:00'),
    autorId: 'user1',
    autorName: 'Sanitäter 1',
    autorRolle: 'Sanitäter',
    kategorie: 'USER',
    titel: 'Erster Eintrag',
    beschreibung: 'Aktiver Eintrag',
    referenzEinsatzId: null,
    referenzPatientId: null,
    referenzEinsatzmittelId: null,
    systemQuelle: null,
    version: 1,
    istAbgeschlossen: false,
    timestampAbschluss: null,
    abgeschlossenVon: null,
    status: EtbEntryDtoStatusEnum.Aktiv
};

// Mock für die Hooks
vi.mock('../../../../hooks/etb/useEinsatztagebuch', () => ({
    useEinsatztagebuch: vi.fn()
}));

vi.mock('../../../../hooks/etb/useFahrzeuge', () => ({
    useFahrzeuge: vi.fn()
}));

// Mock für die Formatierungsfunktion
vi.mock('@/utils/date', () => ({
    formatNatoDateTime: vi.fn(() => '10 JAN 2023 12:00Z')
}));

// Mock für die ETBTable-Komponente
vi.mock('@organisms/etb/ETBTable', () => ({
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    ETBTable: ({ onUeberschreibeEntry }: { onUeberschreibeEntry?: (entry: any) => void }) => (
        <div data-testid="mock-etb-table">
            <button
                data-testid="ueberschreiben-button"
                onClick={() => onUeberschreibeEntry && onUeberschreibeEntry(mockEntryData)}
            >
                Überschreiben
            </button>
        </div>
    )
}));

// Mock für die ETBCardList-Komponente
vi.mock('@organisms/etb/ETBCardList', () => ({
    ETBCardList: () => <div data-testid="mock-etb-cardlist" />
}));

describe('ETBOverview Komponente', () => {
    // Test-Daten
    const mockEntries: EtbEntryDto[] = [
        {
            id: '1',
            laufendeNummer: 1,
            autorId: 'user1',
            autorName: 'Test User',
            timestampErstellung: new Date().toISOString(),
            timestampEreignis: new Date().toISOString(),
            kategorie: 'USER',
            beschreibung: 'Test Eintrag 1',
            status: 'aktiv',
            version: 1,
            istAbgeschlossen: false,
        },
        {
            id: '2',
            laufendeNummer: 2,
            autorId: 'user1',
            autorName: 'Test User',
            timestampErstellung: new Date().toISOString(),
            timestampEreignis: new Date().toISOString(),
            kategorie: 'USER',
            beschreibung: 'Test Eintrag 2',
            status: 'aktiv',
            version: 1,
            istAbgeschlossen: false,
        },
    ];

    const mockFahrzeuge = {
        fahrzeugeImEinsatz: [
            { kennzeichen: 'TEST-123', typ: 'ELW', funkrufname: 'Florian 1' },
        ],
    };

    // Mocks für die Hooks
    const mockUseEinsatztagebuch = {
        einsatztagebuch: {
            data: { items: mockEntries },
            isLoading: false,
            error: null,
            refetch: vi.fn(),
        },
        createEinsatztagebuchEintrag: { mutate: vi.fn(), mutateAsync: vi.fn() },
        archiveEinsatztagebuchEintrag: { mutate: vi.fn(), mutateAsync: vi.fn() },
        ueberschreibeEinsatztagebuchEintrag: { mutate: vi.fn(), mutateAsync: vi.fn().mockResolvedValue({}) },
    };

    const mockUseFahrzeuge = {
        fahrzeuge: { data: mockFahrzeuge, isLoading: false, error: null },
        error: null,
        refreshFahrzeuge: vi.fn(),
    };

    // QueryClient für React Query
    let queryClient: QueryClient;

    beforeEach(() => {
        queryClient = new QueryClient({
            defaultOptions: {
                queries: {
                    retry: false,
                },
            },
        });

        // Mock-Implementierungen setzen
        (useEinsatztagebuch as unknown as ReturnType<typeof vi.fn>).mockReturnValue(mockUseEinsatztagebuch);
        (useFahrzeuge as unknown as ReturnType<typeof vi.fn>).mockReturnValue(mockUseFahrzeuge);

        // Reset mocks between tests
        vi.clearAllMocks();

        // Für den Mobile-Check einen festen Wert zurückgeben (Desktop)
        Object.defineProperty(window, 'innerWidth', {
            writable: true,
            configurable: true,
            value: 1024 // Desktop-Breite
        });
    });

    it('sollte die Tabelle mit Einträgen rendern', () => {
        render(
            <QueryClientProvider client={queryClient}>
                <ETBOverview />
            </QueryClientProvider>
        );

        // Prüfe, ob die Einträge angezeigt werden
        expect(screen.getByText('Test Eintrag 1')).toBeInTheDocument();
        expect(screen.getByText('Test Eintrag 2')).toBeInTheDocument();
    });

    it('sollte einen Ladeindikator anzeigen, wenn Daten geladen werden', () => {
        // Setze isLoading auf true
        (useEinsatztagebuch as unknown as ReturnType<typeof vi.fn>).mockReturnValue({
            ...mockUseEinsatztagebuch,
            einsatztagebuch: {
                ...mockUseEinsatztagebuch.einsatztagebuch,
                isLoading: true,
            },
        });

        render(
            <QueryClientProvider client={queryClient}>
                <ETBOverview />
            </QueryClientProvider>
        );

        // Prüfe, ob der Ladeindikator angezeigt wird
        expect(screen.getByText('Lade Einsatztagebuch...')).toBeInTheDocument();
    });

    it('sollte eine Fehlermeldung anzeigen, wenn ein Fehler auftritt', () => {
        // Setze einen Fehler
        const errorMessage = 'Fehler beim Laden der Daten';
        (useEinsatztagebuch as unknown as ReturnType<typeof vi.fn>).mockReturnValue({
            ...mockUseEinsatztagebuch,
            einsatztagebuch: {
                ...mockUseEinsatztagebuch.einsatztagebuch,
                error: new Error(errorMessage),
            },
        });

        render(
            <QueryClientProvider client={queryClient}>
                <ETBOverview />
            </QueryClientProvider>
        );

        // Prüfe, ob die Fehlermeldung angezeigt wird
        expect(screen.getByText('Fehler beim Laden des Einsatztagebuchs')).toBeInTheDocument();
        expect(screen.getByText(errorMessage)).toBeInTheDocument();
    });

    it('sollte den Überschreibe-Dialog korrekt anzeigen', async () => {
        render(
            <QueryClientProvider client={queryClient}>
                <ETBOverview />
            </QueryClientProvider>
        );

        // Finde den Bearbeiten-Button für den ersten Eintrag
        const editButtons = screen.getAllByRole('button', { name: /überschreiben/i });
        fireEvent.click(editButtons[0]);

        // Warte auf das Öffnen des Dialogs
        await waitFor(() => {
            expect(screen.getByText(/Eintrag 1 von 10 JAN 2023 12:00Z überschreiben/i)).toBeInTheDocument();
        });
    });

    it('sollte den Filter für überschriebene Einträge umschalten', () => {
        render(
            <QueryClientProvider client={queryClient}>
                <ETBOverview />
            </QueryClientProvider>
        );

        // Finde den Switch für überschriebene Einträge
        const toggleSwitch = screen.getByTestId('toggle-ueberschrieben-switch');

        // Prüfe Anfangszustand
        expect(toggleSwitch).not.toBeChecked();

        // Schalte den Switch um
        fireEvent.click(toggleSwitch);

        // Prüfe, ob useEinsatztagebuch mit den richtigen Optionen aufgerufen wurde
        expect(useEinsatztagebuch).toHaveBeenLastCalledWith({ includeUeberschrieben: true });
    });
}); 