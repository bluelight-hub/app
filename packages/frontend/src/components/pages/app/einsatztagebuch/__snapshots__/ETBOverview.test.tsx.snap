// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ETBOverview Komponente > sollte einen Snapshot im Fehlerzustand erstellen 1`] = `
<div>
  <div
    class="px-4 sm:px-6 lg:px-8"
  >
    <div
      class="flex items-center justify-between pb-5 border-b border-gray-200"
    >
      <h1
        class="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate"
      >
        Einsatztagebuch
      </h1>
      <div
        class="mt-3 sm:mt-0 sm:ml-4"
      >
        <div
          class="ant-space css-dev-only-do-not-override-1261szd ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224,200h-8V40a8,8,0,0,0-8-8H152a8,8,0,0,0-8,8V80H96a8,8,0,0,0-8,8v40H48a8,8,0,0,0-8,8v64H32a8,8,0,0,0,0,16H224a8,8,0,0,0,0-16ZM160,48h40V200H160ZM104,96h40V200H104ZM56,144H88v56H56Z"
                  />
                </svg>
              </span>
              <span>
                Dashboard
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224,128a8,8,0,0,1-8,8H136v80a8,8,0,0,1-16,0V136H40a8,8,0,0,1,0-16h80V40a8,8,0,0,1,16,0v80h80A8,8,0,0,1,224,128Z"
                  />
                </svg>
              </span>
              <span>
                Neuer Eintrag
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-4 mb-2"
    >
      <div
        class="ant-space css-dev-only-do-not-override-1261szd ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
      >
        <div
          class="ant-space-item"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-1261szd"
          >
            Überschriebene Einträge anzeigen:
          </span>
        </div>
        <div
          class="ant-space-item"
        >
          <button
            aria-checked="false"
            class="ant-switch css-dev-only-do-not-override-1261szd"
            data-testid="toggle-ueberschrieben-switch"
            role="switch"
            type="button"
          >
            <div
              class="ant-switch-handle"
            />
            <span
              class="ant-switch-inner"
            >
              <span
                class="ant-switch-inner-checked"
              />
              <span
                class="ant-switch-inner-unchecked"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="ant-alert ant-alert-error ant-alert-with-description mb-4 css-dev-only-do-not-override-1261szd"
      data-show="true"
      role="alert"
    >
      <span
        aria-label="close-circle"
        class="anticon anticon-close-circle ant-alert-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="close-circle"
          fill="currentColor"
          fill-rule="evenodd"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
          />
        </svg>
      </span>
      <div
        class="ant-alert-content"
      >
        <div
          class="ant-alert-message"
        >
          Fehler beim Laden des Einsatztagebuchs
        </div>
        <div
          class="ant-alert-description"
        >
          Error: Fehler beim Laden der Daten
        </div>
      </div>
      <div
        class="ant-alert-action"
      >
        <button
          class="px-4 py-2 bg-primary-500 text-white rounded hover:bg-primary-600"
        >
          Erneut versuchen
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ETBOverview Komponente > sollte einen Snapshot im Ladezustand erstellen 1`] = `
<div>
  <div
    class="px-4 sm:px-6 lg:px-8"
  >
    <div
      class="flex items-center justify-between pb-5 border-b border-gray-200"
    >
      <h1
        class="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate"
      >
        Einsatztagebuch
      </h1>
      <div
        class="mt-3 sm:mt-0 sm:ml-4"
      >
        <div
          class="ant-space css-dev-only-do-not-override-1261szd ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224,200h-8V40a8,8,0,0,0-8-8H152a8,8,0,0,0-8,8V80H96a8,8,0,0,0-8,8v40H48a8,8,0,0,0-8,8v64H32a8,8,0,0,0,0,16H224a8,8,0,0,0,0-16ZM160,48h40V200H160ZM104,96h40V200H104ZM56,144H88v56H56Z"
                  />
                </svg>
              </span>
              <span>
                Dashboard
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224,128a8,8,0,0,1-8,8H136v80a8,8,0,0,1-16,0V136H40a8,8,0,0,1,0-16h80V40a8,8,0,0,1,16,0v80h80A8,8,0,0,1,224,128Z"
                  />
                </svg>
              </span>
              <span>
                Neuer Eintrag
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-4 mb-2"
    >
      <div
        class="ant-space css-dev-only-do-not-override-1261szd ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
      >
        <div
          class="ant-space-item"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-1261szd"
          >
            Überschriebene Einträge anzeigen:
          </span>
        </div>
        <div
          class="ant-space-item"
        >
          <button
            aria-checked="false"
            class="ant-switch css-dev-only-do-not-override-1261szd"
            data-testid="toggle-ueberschrieben-switch"
            role="switch"
            type="button"
          >
            <div
              class="ant-switch-handle"
            />
            <span
              class="ant-switch-inner"
            >
              <span
                class="ant-switch-inner-checked"
              />
              <span
                class="ant-switch-inner-unchecked"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="flex justify-center items-center h-64"
    >
      <div
        aria-busy="true"
        aria-live="polite"
        class="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-1261szd"
      >
        <span
          class="ant-spin-dot-holder"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`ETBOverview Komponente > sollte einen Snapshot im Normalzustand erstellen 1`] = `
<div>
  <div
    class="px-4 sm:px-6 lg:px-8"
  >
    <div
      class="flex items-center justify-between pb-5 border-b border-gray-200"
    >
      <h1
        class="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate"
      >
        Einsatztagebuch
      </h1>
      <div
        class="mt-3 sm:mt-0 sm:ml-4"
      >
        <div
          class="ant-space css-dev-only-do-not-override-1261szd ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224,200h-8V40a8,8,0,0,0-8-8H152a8,8,0,0,0-8,8V80H96a8,8,0,0,0-8,8v40H48a8,8,0,0,0-8,8v64H32a8,8,0,0,0,0,16H224a8,8,0,0,0,0-16ZM160,48h40V200H160ZM104,96h40V200H104ZM56,144H88v56H56Z"
                  />
                </svg>
              </span>
              <span>
                Dashboard
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224,128a8,8,0,0,1-8,8H136v80a8,8,0,0,1-16,0V136H40a8,8,0,0,1,0-16h80V40a8,8,0,0,1,16,0v80h80A8,8,0,0,1,224,128Z"
                  />
                </svg>
              </span>
              <span>
                Neuer Eintrag
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-4 mb-2"
    >
      <div
        class="ant-space css-dev-only-do-not-override-1261szd ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
      >
        <div
          class="ant-space-item"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-1261szd"
          >
            Überschriebene Einträge anzeigen:
          </span>
        </div>
        <div
          class="ant-space-item"
        >
          <button
            aria-checked="false"
            class="ant-switch css-dev-only-do-not-override-1261szd"
            data-testid="toggle-ueberschrieben-switch"
            role="switch"
            type="button"
          >
            <div
              class="ant-switch-handle"
            />
            <span
              class="ant-switch-inner"
            >
              <span
                class="ant-switch-inner-checked"
              />
              <span
                class="ant-switch-inner-unchecked"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="mt-8"
    >
      <div
        data-testid="mock-etb-table"
      >
        <div
          data-testid="entries-count"
        >
          2
           Einträge
        </div>
        <div
          data-testid="fahrzeuge-count"
        >
          1
           Fahrzeuge
        </div>
        <div
          data-testid="loading-state"
        >
          Geladen
        </div>
        <button
          data-testid="ueberschreiben-button"
        >
          Überschreiben
        </button>
        <button
          data-testid="archivieren-button"
        >
          Archivieren
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ETBOverview Komponente > sollte einen Snapshot mit geöffnetem Formular erstellen 1`] = `
<div>
  <div
    class="px-4 sm:px-6 lg:px-8"
  >
    <div
      class="flex items-center justify-between pb-5 border-b border-gray-200"
    >
      <h1
        class="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate"
      >
        Einsatztagebuch
      </h1>
      <div
        class="mt-3 sm:mt-0 sm:ml-4"
      >
        <div
          class="ant-space css-dev-only-do-not-override-1261szd ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224,200h-8V40a8,8,0,0,0-8-8H152a8,8,0,0,0-8,8V80H96a8,8,0,0,0-8,8v40H48a8,8,0,0,0-8,8v64H32a8,8,0,0,0,0,16H224a8,8,0,0,0,0-16ZM160,48h40V200H160ZM104,96h40V200H104ZM56,144H88v56H56Z"
                  />
                </svg>
              </span>
              <span>
                Dashboard
              </span>
            </button>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224,128a8,8,0,0,1-8,8H136v80a8,8,0,0,1-16,0V136H40a8,8,0,0,1,0-16h80V40a8,8,0,0,1,16,0v80h80A8,8,0,0,1,224,128Z"
                  />
                </svg>
              </span>
              <span>
                Neuer Eintrag
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-4 mb-2"
    >
      <div
        class="ant-space css-dev-only-do-not-override-1261szd ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
      >
        <div
          class="ant-space-item"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-1261szd"
          >
            Überschriebene Einträge anzeigen:
          </span>
        </div>
        <div
          class="ant-space-item"
        >
          <button
            aria-checked="false"
            class="ant-switch css-dev-only-do-not-override-1261szd"
            data-testid="toggle-ueberschrieben-switch"
            role="switch"
            type="button"
          >
            <div
              class="ant-switch-handle"
            />
            <span
              class="ant-switch-inner"
            >
              <span
                class="ant-switch-inner-checked"
              />
              <span
                class="ant-switch-inner-unchecked"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="mt-4 p-4 border border-gray-200 rounded-md shadow-sm"
    >
      <div
        class="flex justify-between items-center mb-2"
      >
        <h3
          class="text-lg font-medium text-gray-900"
        >
          Neuer Einsatztagebucheintrag
        </h3>
        <button
          class="ant-btn css-dev-only-do-not-override-1261szd ant-btn-text ant-btn-color-default ant-btn-variant-text ant-btn-icon-only text-gray-500 hover:text-gray-700"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
              />
            </svg>
          </span>
        </button>
      </div>
      <p
        class="text-sm text-gray-500 mb-4"
      >
        Bitte füllen Sie alle Felder aus
      </p>
      <div
        data-testid="mock-etb-entry-form"
      >
        <div
          data-testid="form-mode"
        >
          Create Mode
        </div>
        <button
          data-testid="form-submit-button"
        >
          Formular Absenden
        </button>
        <button
          data-testid="form-cancel-button"
        >
          Abbrechen
        </button>
      </div>
    </div>
    <div
      class="mt-8"
    >
      <div
        data-testid="mock-etb-table"
      >
        <div
          data-testid="entries-count"
        >
          2
           Einträge
        </div>
        <div
          data-testid="fahrzeuge-count"
        >
          1
           Fahrzeuge
        </div>
        <div
          data-testid="loading-state"
        >
          Geladen
        </div>
        <button
          data-testid="ueberschreiben-button"
        >
          Überschreiben
        </button>
        <button
          data-testid="archivieren-button"
        >
          Archivieren
        </button>
      </div>
    </div>
  </div>
</div>
`;
